### PR TITLE
Reusable and callable brassy workflow

### PR DESCRIPTION
**Pull request by biosafetylvl5**
* [x] NO REQUIRED ***existing tests*** (explain why not required)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] NO REQUIRED ***integration tests*** (explain why not required)
* [x] NO REQUIRED ***documentation*** (explain why not required)
* [x] Required ***release notes*** added for new/modified functionality
* [x] Required ***updates to other repos*** complete (see https://github.com/NRLMMD-GEOIPS/geoips_clavrx/pull/41 for an example)

# Testing Instructions
Go to https://github.com/NRLMMD-GEOIPS/geoips_clavrx/pull/41 to test.

# Summary

This migrates a workflow from https://github.com/NRLMMD-GEOIPS/geoips/pull/746 into this repo in a callable format so other repos can use it.

It also integrates the changes from https://github.com/NRLMMD-GEOIPS/geoips/pull/761 to tag and re-tag commits.

**THIS MUST BE MERGED BEFORE MERGING CALLING WORKFLOWS**

### CODE REVIEW

File: .github/workflows/generate-release-note.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Extract branch name
+        id: extract_branch
+        run: |
+          git rev-parse --abbrev-ref HEAD
+          echo ${{ github.ref }}
+          export raw=$(git branch -r --contains ${{ github.ref }})
+          echo $raw
+          branch=${raw##*/}
+          echo "branch=${branch}" >> $GITHUB_OUTPUT
+          echo "Working branch is $branch"
```
Comment (jsolbrig) [inline .github/workflows/generate-release-note.yaml:25]: I think I remember that it doesn't work, but couldn't we just use `${{ github.ref_name }}`? Note, this wouldn't work in the case where the workflow is triggered by pushing a new tag, but it should work here. Maybe not worth changing.

File: .github/workflows/generate-release-note.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+      - name: Extract version
+        id: extract_version
+        run: |
+          export latest_tag=`git -C . tag --sort=-creatordate | grep -v osr | head -n 1`
+          echo "Version is $latest_tag"
+          echo "version=${latest_tag}" >> $GITHUB_OUTPUT
```
Comment (jsolbrig) [inline .github/workflows/generate-release-note.yaml:32]: What would happen here if we had a non-version tag on the repo? E.g. `jsolbrigs-test-tag`.

**Review by jsolbrig — APPROVED** [review]

File: .github/workflows/generate-release-note.yaml (hunk 1 above)
Comment (biosafetylvl5) [reply .github/workflows/generate-release-note.yaml:25]: You're right - the `ref_name` would be ideal here. However, when it's called on-tag, the ref is the tag. Hence why I do it this way here

Comment (biosafetylvl5) [reply .github/workflows/generate-release-note.yaml:25]: It is ugly! But it works 🔮

File: .github/workflows/generate-release-note.yaml (hunk 2 above)
Comment (biosafetylvl5) [reply .github/workflows/generate-release-note.yaml:32]: It would run with that tag :)

Comment (biosafetylvl5) [reply .github/workflows/generate-release-note.yaml:32]: I leave filtering up to the calling workflow, as this isn't meant to be run directly.

Comment (jsolbrig) [reply .github/workflows/generate-release-note.yaml:32]: Sounds good. We should make sure to document that fact.

File: .github/workflows/generate-release-note.yaml (hunk 1 above)
Comment (jsolbrig) [reply .github/workflows/generate-release-note.yaml:25]: Sounds good.

**Review by mindyls — APPROVED** [review]

**Comment by biosafetylvl5** [comment]
Closed as implemented internally at NRL

**Pull request closed without merging**